### DISCUSSION TITLE
[Feature request] List of colors for GroupBox widget

### DISCUSSION THREAD

**Discussion by Crystal4276** [qtile/qtile#4015] [Ideas]
This screenshot is from the custom configuration of @jx11r (https://github.com/jx11r/qtile): 
![image](https://user-images.githubusercontent.com/27446196/202996379-2a757b69-d439-4899-80b3-ab4e9b840781.png)

I find the general look of the GroupBox widget really pleasing when **labels** and **highlight methods** can take different colors.

I saw that kind of results with others WMs too, so I was wondering if Qtile could have it implemented ?
If we could specify a list of colors associated to the number of groups that would be super nice.

**Comment by elParaguayo** [comment]
Well, it's definitely possible. If you've got it working already, do you want to submit a pull request?

**Reply by Crystal4276** [reply]
> Well, it's definitely possible. If you've got it working already, do you want to submit a pull request?

I don't have it working already. 
I could try "stealing"  jx11r code and make it work on my config.
Tbh, Qtile is my very first experience with coding. 
I just learned the basics of python one month ago, it feels a bit overwhelming/suicidal to try making a PR.

I have no idea what it would take to reach something decent (and i would need to learn how to use github).
Any thoughts on what's realistic (given my current knowledge)?

**Reply by jx11r** [reply]
Or I could open that PR, no problem for me.

**Reply by Crystal4276** [reply]
> Or I could open that PR, no problem for me.

I think i will leave it to you. I'd love to see this implemented in a not so distant future. 
If i try it will probably take me several weeks.

I guess it's time for me to learn more skills to be able also to contribute (at least for small stuffs)...

**Reply by elParaguayo** [reply]
I'm happy to look at the PR whoever submits it!

**Reply by jx11r** [reply]
> I'd love to see this implemented in a not so distant future.

Unfortunately, the GroupBox widget needs some improvements first, and it might take some time.
So I can't submit my pull request right now because of that, but I will be working on it :)

**Reply by elParaguayo** [reply]
Sometimes I feel it needs a complete rewrite...

**Reply by jx11r** [reply]
Yes, I think so too.

**Reply by Crystal4276** [reply]
> So I can't submit my pull request right now because of that, but I will be working on it :)

Thanks a lot !

**Comment by elParaguayo** [comment]
Move to #4365
